src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.cs

```diff
@@ -337,7 +338,7 @@ internal bool IsSupportedDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diag
 
                 // Editorconfig user settings override compilation wide settings.
                 if (isSuppressed &&
-                    isEnabledWithAnalyzerConfigOptions(diag, severityFilter, analyzerExecutor.Compilation, analyzerExecutor.AnalyzerOptions))
+                    isEnabledWithAnalyzerConfigOptions(diag, severityFilter, compilation, analyzerOptions))
```

Comment:
😟 This one leaves me with a bit of an uneasy feeling. Likely worth a pass to make sure there are no remaining uses of the two original properties where null is a possibility.

Reply:
Yep, I identified this via a failing test - this is the only path where we use an AnalyzerExecutor created without a compilation - basically needed to support the below public API based on options, the only API where analyzer can be called without a compilation: https://github.com/dotnet/roslyn/blob/0d3cf92c300459e40dae31b09aa6c7f5001aaaa4/src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs#L1319-L1323